test/e2e/nodepool_nto_machineconfig_test.go

```diff
@@ -46,222 +50,335 @@ spec:
+		err := mgmtClient.List(ctx, nodePools, &crclient.ListOptions{
+			Namespace: hostedCluster.Namespace,
+		})
+		g.Expect(err).NotTo(HaveOccurred(), "failed getting existant nodepools")
+		for _, nodePool := range nodePools.Items {
+			if !strings.Contains(nodePool.Name, "-test-") {
+				originalNP = nodePool
+			}
 		}
-		nodePool.Spec.Management.Replace = &hyperv1.ReplaceUpgrade{
-			Strategy: hyperv1.UpgradeStrategyRollingUpdate,
-			RollingUpdate: &hyperv1.RollingUpdate{
-				MaxUnavailable: func(v intstr.IntOrString) *intstr.IntOrString { return &v }(intstr.FromInt(0)),
-				MaxSurge:       func(v intstr.IntOrString) *intstr.IntOrString { return &v }(intstr.FromInt(int(*nodePool.Spec.Replicas))),
+		g.Expect(originalNP.Name).NotTo(BeEmpty())
+		g.Expect(originalNP.Name).NotTo(ContainSubstring("test"))
+		awsNPInfo := originalNP.Spec.Platform.AWS
+
+		// Define a new Nodepool
+		nodePool := &hyperv1.NodePool{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "NodePool",
+				APIVersion: hyperv1.GroupVersion.String(),
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      hostedCluster.Name + "-" + "test-ntomachineconfig-replace",
+				Namespace: hostedCluster.Namespace,
+			},
+			Spec: hyperv1.NodePoolSpec{
+				Management: hyperv1.NodePoolManagement{
+					UpgradeType: hyperv1.UpgradeTypeReplace,
+					AutoRepair:  true,
+					Replace: &hyperv1.ReplaceUpgrade{
+						Strategy: hyperv1.UpgradeStrategyRollingUpdate,
+						RollingUpdate: &hyperv1.RollingUpdate{
+							MaxUnavailable: func(v intstr.IntOrString) *intstr.IntOrString { return &v }(intstr.FromInt(0)),
+							MaxSurge:       func(v intstr.IntOrString) *intstr.IntOrString { return &v }(intstr.FromInt(int(oneReplicas))),
+						},
+					},
+				},
+				ClusterName: hostedCluster.Name,
+				Replicas:    &oneReplicas,
+				Release: hyperv1.Release{
+					Image: hostedCluster.Spec.Release.Image,
+				},
+				Platform: hyperv1.NodePoolPlatform{
+					Type: hostedCluster.Spec.Platform.Type,
+					AWS:  awsNPInfo,
+				},
 			},
 		}
-	}
-
-	hostedCluster := e2eutil.CreateCluster(t, ctx, client, &clusterOpts, globalOpts.Platform, globalOpts.ArtifactDir)
-
-	// Sanity check the cluster by waiting for the nodes to report ready
-	t.Logf("Waiting for guest client to become available")
-	guestClient := e2eutil.WaitForGuestClient(t, testContext, client, hostedCluster)
+		// Create NodePool for current test
+		err = mgmtClient.Create(ctx, nodePool)
+		if err != nil {
+			if !errors.IsAlreadyExists(err) {
+				t.Fatalf("failed to create nodePool %s with Autorepair function: %v", nodePool.Name, err)
+			}
+			err = nodePoolRecreate(t, ctx, nodePool, mgmtClient)
+			g.Expect(err).NotTo(HaveOccurred(), "failed to Create the NodePool")
+		}
+		defer nodePoolScaleDownToZero(ctx, mgmtClient, *nodePool, t)
 
-	// Wait for Nodes to be Ready
-	e2eutil.WaitForNReadyNodes(t, testContext, guestClient, clusterOpts.NodePoolReplicas, hostedCluster.Spec.Platform.Type)
+		numNodes := int32(1)
```

Comment:
why is this 1 now? what was is before? I'd expect it to be at least 2

Reply:
only in the Replace case? or also for InPlace?

Reply:
Modified in both cases (InPlace and Replace)